modules/addNavigation.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * History enhancer that adds additional navigation methods — `transitionTo()`
+ * and `replaceWith()`. TODO: move to history module (?)
+ * @param {CreateHistory} createHistory - History-creating function
+ * @returns {CreateHistory}
+ */
+export default function addNavigation(createHistory) {
+  return (...args) => {
+    const history = createHistory(...args);
+
+    /**
+     * Pushes a new Location onto the history stack.
+     * @param {string} pathname
+     * @param {Object} query
+     * @param {Object} State
+     */
+    function transitionTo(pathname, query, state = null) {
+      history.pushState(state, history.createHref(pathname, query));
```

Comment:
`createHref` isn't quite what we want here. What we really want is `createPath`, but there's really no reason to make that public API in the history module since the only consumer would probably be the router.

What I'd really like to do is force users to use [`enableQueries`](https://github.com/rackt/history/blob/master/modules/enableQueries.js) if they want custom query parsing/serialization.